src/buffers/Buffer.cpp

```diff
@@ -37,12 +38,12 @@ bool Buffer::Open(const std::string inputUrl, int optFlag)
     {
       ss << inputUrl;
     }
-    m_inputHandle = XBMC->OpenFile(ss.str().c_str(), optFlag );
+    kodi::vfs::CFile m_inputHandle;
```

Comment:
With you file open, as you added `kodi::vfs::CFile` is it away after end of Open() call, This need to place inside a header.

Reply:
Thanks @phunkyfish lead me to this too I had two reference to the variable.